src/composables/useMessageInfo.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * @copyright Copyright (c) 2024 Dorra Jaouad <dorra.jaoued1@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { computed } from 'vue'
+
+import { getCapabilities } from '@nextcloud/capabilities'
+import moment from '@nextcloud/moment'
+
+import { useConversationInfo } from './useConversationInfo.js'
+import { useStore } from './useStore.js'
+import { PARTICIPANT } from '../constants.js'
+
+/**
+ * Check whether the user can edit the message or not
+ *
+ * @param {string} token conversation token
+ * @param {string} messageId message id to edit
+ *
+ * @return {import('vue').ComputedRef<boolean>}
+ */
+export function useMessageInfo(token = null, messageId = null) {
+	const store = useStore()
+
+	// Get the conversation and message
+	const conversation = computed(() => store.getters.conversation(token))
+	const message = computed(() => store.getters.message(token, messageId))
+
+	const {
+		isOneToOne,
+		isConversationReadOnly,
+	} = useConversationInfo({ item: conversation })
+
+	const isModifiable = computed(() =>
+		!isConversationReadOnly.value
+		&& conversation.value.participantType !== PARTICIPANT.TYPE.GUEST)
+
+	const isObjectShare = computed(() => Object.keys(Object(message.value.messageParameters)).some(key => key.startsWith('object')))
+
+	const isMyMsg = computed(() =>
+		message.value.actorId === store.getters.getActorId()
+		&& message.value.actorType === store.getters.getActorType()
+	)
+
+	const canEditMessage = getCapabilities()?.spreed?.features?.includes('edit-messages')
```

Comment:
That's better to move out of composable (after import), so it gets called only once